### PR TITLE
Coverage of hiredis.c

### PR DESCRIPTION
**Pull request by ashtul**
In hiredis.c, some of the uncovered functions and lines are likely to be dead.

### CODE REVIEW

**Comment by ashtul** [comment]
@yossigo 
This is a very small commit but...
A. It covers all of hiredis.c (not including blocking functions)
B. I would like to see the process of getting PRs in.

File: hiredis.c [hunk 1]
```diff
@@ -122,10 +122,10 @@ static void *createStringObject(const redisReadTask *task, char *str, size_t len
         return NULL;
 
     buf = malloc(len+1);
-    if (buf == NULL) {
+    if (buf == NULL) {          // LCOV_EXCL_START
```
Comment (yossigo) [inline hiredis.c]: @ashtul Did you consider the tradeoff between more accurate coverage and code clutter ?
Not sure it's worth having LCOV-specific tags all over the place.

File: test.c [hunk 2]
```diff
@@ -452,13 +472,25 @@ static void test_reply_reader(void) {
     /* Regression test for issue #45 on GitHub. */
     test("Don't do empty allocation for empty multi bulk: ");
     reader = redisReaderCreate();
-    redisReaderFeed(reader,(char*)"*0\r\n",4);
+    redisReaderFeed(reader,(char*)"*0\r\n",8);
```
Comment (yossigo) [inline test.c]: I think this is wrong, the length of the string is 4 bytes.

File: hiredis.c (hunk 1 above)
Comment (ashtul) [reply hiredis.c]: I see your point
This is really your call
I will remove

**Comment by ashtul** [comment]
@yossigo Shell we add CodeCov tag?

**Comment by michael-grunder** [comment]
There's good stuff in here that we should probably get merged before v1.0.0

I went ahead and rebased your PR against the new pluggable-allocation branch to make eventual merging easier.

[Link to rebase](https://github.com/michael-grunder/hiredis/tree/gcov-rebase)

**Pull request closed without merging**